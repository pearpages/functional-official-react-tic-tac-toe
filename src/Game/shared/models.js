const PLAYER1 = 'X';
const PLAYER2 = 'O';

export const PLAYERS = {
  PLAYER1,
  PLAYER2,
};

export const LINES = [
  [0, 1, 2],
  [3, 4, 5],
  [6, 7, 8],
  [0, 3, 6],
  [1, 4, 7],
  [2, 5, 8],
  [0, 4, 8],
  [2, 4, 6],
];
