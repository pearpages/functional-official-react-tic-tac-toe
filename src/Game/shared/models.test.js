import { LINES, PLAYERS } from './models';

describe('models', () => {
  test('When LINES change', () => {
    expect(LINES).toMatchSnapshot();
  });
  test('When PLAYERS change', () => {
    expect(PLAYERS).toMatchSnapshot();
  });
});
