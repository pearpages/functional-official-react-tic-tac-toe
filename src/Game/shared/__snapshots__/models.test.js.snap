// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models When LINES change 1`] = `
Array [
  Array [
    0,
    1,
    2,
  ],
  Array [
    3,
    4,
    5,
  ],
  Array [
    6,
    7,
    8,
  ],
  Array [
    0,
    3,
    6,
  ],
  Array [
    1,
    4,
    7,
  ],
  Array [
    2,
    5,
    8,
  ],
  Array [
    0,
    4,
    8,
  ],
  Array [
    2,
    4,
    6,
  ],
]
`;

exports[`models When PLAYERS change 1`] = `
Object {
  "PLAYER1": "X",
  "PLAYER2": "O",
}
`;
